### PR TITLE
net: ipv6: Add APIs to check ipv6 address is link local and global

### PR DESCRIPTION
**Pull request by sandip-dalvi**

### CODE REVIEW

File: include/zephyr/net/net_ip.h [hunk 1]
```diff
@@ -853,6 +866,18 @@ static inline bool net_ipv6_is_ula_addr(const struct in6_addr *addr)
...
+/**
+ * @brief Check if the given IPv6 address is a global address.
+ *
+ * @param addr A valid pointer on an IPv6 address
+ *
+ * @return True if it is, false otherwise.
+ */
+static inline bool net_ipv6_is_global_addr(const struct in6_addr *addr)
+{
+	return addr->s6_addr[0] == 0x20;
```
Comment (rlubos) [inline include/zephyr/net/net_ip.h:878]: Shoudl `addr->s6_addr[0]` be ored with `0xE0` before comparing? Only the first 3 bits indicate address scope, current logic will incorrectly filter out global addresses other than thise starting with `0x20`.

File: include/zephyr/net/net_ip.h [hunk 2]
```diff
@@ -841,6 +841,19 @@ static inline bool net_ipv6_is_ll_addr(const struct in6_addr *addr)
 	return UNALIGNED_GET(&addr->s6_addr16[0]) == htons(0xFE80);
 }
 
+/**
+ * @brief Check if the given IPv6 address is a link local address.
```
Comment (rlubos) [inline include/zephyr/net/net_ip.h:845]: `site local`

**Review by rlubos — COMMENTED** [review]
The commit message needs some body.

**Comment by dleach02** [comment]
@sandip-dalvi please review https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-message-guidelines for guidance on your commit message.

File: include/zephyr/net/net_ip.h (hunk 1 above)
Comment (sandip-dalvi) [reply include/zephyr/net/net_ip.h:878]: I think we have to use and and not or.

File: include/zephyr/net/net_ip.h (hunk 2 above)
Comment (sandip-dalvi) [reply include/zephyr/net/net_ip.h:845]: Okay

File: include/zephyr/net/net_ip.h (hunk 1 above)
Comment (rlubos) [reply include/zephyr/net/net_ip.h:878]: Of course, you're right, I don't know what I was thinking. We need to &.

Comment (sandip-dalvi) [reply include/zephyr/net/net_ip.h]: Done

File: include/zephyr/net/net_ip.h (hunk 2 above)
Comment (sandip-dalvi) [reply include/zephyr/net/net_ip.h]: Updated

**Review by rlubos — APPROVED** [review]

**Review by pdgendt — APPROVED** [review]

File: include/zephyr/net/net_ip.h [hunk 3]
```diff
@@ -841,6 +841,19 @@ static inline bool net_ipv6_is_ll_addr(const struct in6_addr *addr)
 	return UNALIGNED_GET(&addr->s6_addr16[0]) == htons(0xFE80);
 }
 
+/**
+ * @brief Check if the given IPv6 address is a site local address.
```
Comment (jukkar) [inline include/zephyr/net/net_ip.h:845]: Commit message talks about "link local" so it should be changed to "site local"?

**Review by jukkar — CHANGES_REQUESTED** [review]

File: include/zephyr/net/net_ip.h (hunk 3 above)
Comment (sandip-dalvi) [reply include/zephyr/net/net_ip.h:845]: Updated

**Review by jukkar — APPROVED** [review]

**Comment by sandip-dalvi** [comment]
> The commit message needs some body.

Updated

**Comment by jukkar** [comment]
@sandip-dalvi could you rebase the PR against latest upstream and force push, there is some weird CI issue that hopefully solves it

**Comment by sandip-dalvi** [comment]
> @sandip-dalvi could you rebase the PR against latest upstream and force push, there is some weird CI issue that hopefully solves it

It is done

**Review by pdgendt — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Pull request merged**